base/stream.jl

```diff
@@ -946,11 +946,13 @@ for (x, writable, unix_fd, c_symbol) in
                 handle.handle)
             handle
         end
+        function ($f)(handle::Pipe)
+            link_pipe(handle, julia_only_read = $(writable), julia_only_write = $(!writable))
```

Comment:
probably needs to check whether the Pipe is already open, following logic from https://github.com/JuliaLang/julia/blob/f06840ee80f70b6823464a35a4b9b9f78254cb8b/base/process.jl#L410-L412